Issue #18: Add Global Blacklist & whitelist to apply

bluesven869 opened the pull request:
- Currently, Blacklist & Whiltelist UI & Logic have been implemented.

Commit:
Force SSL only access.

Commit:
Tweak filter boxes css to wrap text.  This may not be the most elegant solution but it makes the longer descriptions work.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Added guzzle as a requirement.

Commit:
Added support to send out emails on user deactivations.  These emails are very generic so far.

Commit:
Move update files to under views.

Commit:
Setup routes to provide dynamically generate appcasts.

Commit:
Renamed email view to keep naming consistent.

Commit:
Added support to send emails when a deactivation request is granted.

Adjusted deactivation request received email to provide more information.

Commit:
Updated app version and appcast files.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Add Job and migrations for jobs.  Add api_token migration.

Commit:
Add requirements for user migration changes.
More work on adding support for a restful api.

Commit:
Updated appcast version .

Commit:
Added support to log sql queries by uncommenting code in Providers/AppServiceProvider.php

Commit:
Added doctrine/dbal requirement to support editing database columns.

Commit:
Removed api_token migration as we're not going that route.

Commit:
Add functioning basic authentication to management portion of api.

Commit:
Starting cleaning up old token support and adding basic auth support.

Commit:
Updated distributed version.

Commit:
Added SQL logging support.

Commit:
Merge branch 'master' of https://github.com/cloudveiltech/CitadelManager

Commit:
Add Laravel Passport for API token authentication.

As part of this I removed the default middleware group from the api file
to allow us to use multiple types of authentication in that file.  Please see
the notes in routes/api.php.

This commit requires a few command line arguments to function properly.
1. php artisan migrate
2. php artisan passport:install

One app sign in it should issue a GET to api/user/gettoken with basic authentication.
This will return a long lived (100 year) token.  This token can be revoked in the database but
we don't currently have a ui to do this.  It can be revoked by the app as explained below.

You can retrieve user info with this token by doing a GET to api/v2/me/user using header authentication.

Once the app is done with the token it can do a POST to /api/v2/revoketoken and the current token
will be revoked.

Commit:
Remove remnants of the api_token functionality from the User model.  Corrected adding users functions.  closes 3.

Commit:
Added current version of composer.lock.

TechnikEmpire committed:
Fixes an issue where filter lists embedded in the root zip directory create categories
that are named after the PHP tmp zip file. Such zip entries are now ignored.

Fixes an issue where the overwrite command on uploads caused categories to be purged
more than once.

Converts the user api gettoken into a basic auth API that will properly increment
the license counts of users that authenticate through that route.

Adds user v2 api routes to the CSRF token middleware exception list.

Converts the v2 user gettoken route into a POST route from GET.

Add zip files in the storage dir to the gitignore list. Ignores tmp zip data uploads.

Commit:
Updated distributable version of binary to 21

Commit:
Resolved #15

Added support to create users via basic auth for integration with 3rd party systems.
Added support to search for specific users and retrieve all users via basic auth.

Commit:
Manually revoke all tokens on disabling or deleting a user.
Close #13

Commit:
Wrapped events that send emails in try/catch and sent output to error log.
Close #14

Commit:
Update build version.

Commit:
Close Issue #2

Add support to have users debugging tools upload logs to the controller.  At this point we don't notify tech support or do anything other than store the files.

Commit:
Close Issue #19

Commit:
Add /norestart flag to updater.

Commit:
Started moving release management into its own controller.

Commit:
Added support for admin users to retrieve the list of activations per user via the api.

Commit:
Started adding support to lookup activations in the admin page.   See Issue #21.

Commit:
Close Issue #17.  The setting can be controlled from config/app.php or set in .ENV.

Commit:
Corrected schedule of deactivation requests deletion.

Commit:
Close Issue #20.

Commit:
Altered User model regarding Issue #23.  Needs ui changes yet.

TechnikEmpire committed:
Added three, hard-coded update channels to group configuration.

Commit:
Upgrade to version 1.3.1.
Finish adding support to have Alpha/Beta/Stable release channels.

Close issue #5.

Commit:
Altered outgoing mail to include instructions and added replyto on emails.

Commit:
Added replyto on emails.

Commit:
Added support for basic auth api to edit users information.

Commit:
Corrected code to save last contact timestamps for activations.

Close issue #19

Commit:
Updated code to handle update channels.
Close Issue #5

Commit:
Add jquery and typescript as NPM dependencies.

Commit:
Added support to the app_user_activations table to store the ip address the installation is calling in from.

Commit:
Correct app_user_activation controller to not return an error when no activations are found.

Commit:
issue-18: Added Global White & Black list

TechnikEmpire commented:
@bluesven869 Sorry, who are you? I'm curious because you're trying to get me to merge in changes from the cloudveil fork here?? Can I ask why?

TechnikEmpire commented:
I'm thinking this PR was meant to be applied to the cloudveil fork, not this repo.

Commit:
1. Getting All Datas
 - GetBlackList
 - GetWhiteList
 - GetGroupList
2. Adjustment of Groups
 - MoveToRightAll
 - MoveToRight
 - MoveToLeft
 - MoveToLeftAll

Commit:
## issue 18 completed
 - AddToApply to Specific group(s).

bluesven869 committed:
Merge pull request #31 from cloudveiltech/luis

issue-18: Added Global White & Black list

TechnikEmpire commented:
Closing because these are not meant for this repo, they're for the cloudveil fork.
